napari/components/viewer_model.py

```diff
@@ -580,28 +580,63 @@ def _calc_status_from_cursor(
     ) -> tuple[str | Dict, str] | None:
         if not self.mouse_over_canvas:
             return None
-        active = self.layers.selection.active
-        if active is not None and active._loaded:
-            status = active.get_status(
+        coord2val: dict[str, list[str]] = {}
+        coord_str = ''
+        status_str = ''
+        tooltip_text = ''
+        selection = self.layers.selection
+        active = selection.active
+
+        for layer in self.layers[::-1]:
+            if (
+                not layer.visible
+                or layer.opacity == 0
+                or not layer._loaded
+                or (layer not in selection and not self.grid.enabled)
+            ):
+                continue
+            status = layer.get_status(
                 self.cursor.position,
                 view_direction=self.cursor._view_direction,
                 dims_displayed=list(self.dims.displayed),
                 world=True,
             )
-
-            if self.tooltip.visible:
-                tooltip_text = active._get_tooltip_text(
-                    np.asarray(self.cursor.position),
-                    view_direction=np.asarray(self.cursor._view_direction),
-                    dims_displayed=list(self.dims.displayed),
-                    world=True,
+            separator = '    '
+            emphasis = separator if layer is active else ''
+            coord_str = f'{status["coords"]} » '
+            if status['value'] != '':
+                if coord_str not in coord2val:
+                    coord2val[coord_str] = []
+                coord2val[coord_str].append(
+                    f'{layer.name}: {status["value"]}{emphasis}'
                 )
-            else:
-                tooltip_text = ''
+        if coord2val:
+            if not self.grid.enabled:
+                # use a single coordinate system
+                values = list(itertools.chain(*coord2val.values()))
+                key = next(iter(coord2val))  # choose arbitrary coordinate
+                coord2val = {key: values}
+            status_strs = [
+                key + separator.join(values)
+                for key, values in coord2val.items()
+            ]
+            status_str = separator.join(status_strs)
+        elif coord_str and not self.grid.enabled:
+            status_str = coord_str + '[empty]'
+        elif self.grid.enabled:
+            status_str = '[empty]'
+        else:
+            status_str = 'Ready'
 
-            return status, tooltip_text
+        if self.tooltip.visible and active is not None and active._loaded:
+            tooltip_text = active._get_tooltip_text(
+                np.asarray(self.cursor.position),
+                view_direction=np.asarray(self.cursor._view_direction),
+                dims_displayed=list(self.dims.displayed),
+                world=True,
+            )
 
-        return 'Ready', ''
+        return status_str, tooltip_text
```

Comment:
Here I changed return from dict to a string; it was unintentional to change it always, rather than only in grid and multiple-selection modes. Will update in upcoming PR.